docs/backend-system/architecture/05-extension-points.md

```diff
@@ -25,6 +29,54 @@ export const scaffolderActionsExtensionPoint =
   });
 ```
 
-#### Registering an Extension Point
+## Registering an Extension Point
+
+For modules to be able to use your extension point, and implementation of it must be registered by the plugin. This is done using the `registerExtensionPoint` method in the `register` callback of the plugin definition.
+
+```ts
+class ActionsExtension implements ScaffolderActionsExtensionPoint {
+  addActions(...actions: TemplateAction<any>[]): void { ... }
+
+  getRegisteredActions() { ... }
+}
+
+export const scaffolderPlugin = createBackendPlugin(
+  {
+    id: 'scaffolder',
+    register(env) {
+      const actionsExtensions = new ActionsExtension();
+      env.registerExtensionPoint(
+        scaffolderActionsExtensionPoint,
+        actionsExtensions,
+      );
+
+      env.registerInit({
+        deps: { ... },
+        async init({ ... }) {
+          const actions = actionsExtension.getRegisteredActions();
+
+          // Use the registered actions when setting up the scaffolder ...
+        },
+      });
+    },
+  },
+);
+```
+
+There are a couple of things to note here. The first is that our `ActionsExtension` class both implements the `ScaffolderActionsExtensionPoint` interface, but also has additional public methods. These methods won't be available to the modules that use this extension, but we can use them here in the implementation of our plugin.
+
+The second is that we create our `ActionsExtension` instance within the `register` method, and then access it directly in our `init` method. This is both safe to do and an intended convenience. All modules that extend our plugin will be completely initialized before our plugin gets initialized, which means that at the point where our `init` method is called, all actions have been added and can be accessed.
+
+## Extension Point Design
+
+Designing the extension point interface can be quite challenging. It is a public API surface that your plugin will need to maintain over time. Keep in mind that the installation of modules is an intentional action by the user, meaning you can always design an extension point interface for additions only. For example, there is no need for the `scaffolderActionsExtensionPoint` to support removal of actions, since the user can just uninstall the module that added the action.
```

Comment:
```suggestion
Designing the extension point interface requires careful consideration. It is a public API surface that your plugin will need to maintain over time. Keep in mind that the installation of modules is an intentional action by the user, meaning you can always design an extension point interface for additions only. For example, there is no need for the `scaffolderActionsExtensionPoint` to support removal of actions, since the user can just uninstall the module that added the action.
```